crates/sqlexec/src/context.rs

```diff
@@ -336,6 +336,22 @@ impl SessionContext {
         Ok(())
     }
 
+    pub async fn delete(&mut self, plan: Delete) -> Result<()> {
+        let (database, schema, name) = self.resolve_table_ref(plan.table_name)?;
+
+        if let Some(table_entry) = self
+            .metastore_catalog
+            .resolve_native_table(&database, &schema, &name)
+        {
+            self.tables
+                .delete_rows_where(table_entry, plan.expr)
+                .await?
+        } else {
+            return Err(ExecError::MissingObject { typ: "table", name });
```

Comment:
Minor, but would prefer an error that says something along the lines of "Deleting from external tables not currently supported"